### ISSUE TITLE
Detekt final report name is weird

### ISSUE THREAD

**Issue by Writtscher** [detekt/detekt#3926]
Hi. Not sure whether this is the correct behaviour (I don't think it is). When I register a custom source set 

```
val integrationTest = sourceSets.create("integrationTest") {
  java.srcDir("src/integrationTest/kotlin")
  resources.srcDir("src/integrationTest/resources")
  compileClasspath += sourceSets.main.get().output + sourceSets.test.get().output
  runtimeClasspath += sourceSets.main.get().output + sourceSets.test.get().output
}

configurations {
  ...
  getByName(integrationTest.implementationConfigurationName).extendsFrom(configurations.testImplementation.get())
  getByName(integrationTest.runtimeOnlyConfigurationName).extendsFrom(configurations.testRuntimeOnly.get())
}
```

The final report name is `integrationTest.html` instead of `main.html` even though integration test is not part of the check
```
tasks {
  detekt {
    input.setFrom(DEFAULT_SRC_DIR_KOTLIN)
  }
}
```

**Comment by cortinico** [comment]
> The final report name is `integrationTest.html` instead of `main.html` even though integration test is not part of the check

Sadly we don't have enough information to support here.
Specifically we need to understand:
1. Which task are you invoking from the command line?
2. Are you on a JVM/Android/Multiplatform project?
3. Could you provide a reproducer or share your build.gradle in some form?

**Comment by Writtscher** [comment]
@cortinico No problem. I'll provide a simple project.

**Comment by Writtscher** [comment]
There you go:

https://github.com/Writtscher/detekt-sample

I found out when I remove this (had this in a legacy project and we migrated it to java 11)

```
tasks {
  withType<Detekt> {
    jvmTarget = "11"
  }
}
```

Everything works as expected. Adding it breaks it or at least the behaviour is not as intended. Also the "UP-TO-DATE"-Check does not work accordingly.

**Comment by cortinico** [comment]
Thanks for the reproducer. It looks like a bug of some sort that we should look into.
I was able to reproduce but I could not say what's the root cause at this stage.

**Comment by Writtscher** [comment]
It is not a major issue. I can get along it since it is a legacy project. Nevertheless I think it is not intended to be like that.

**Comment by cortinico** [comment]
So I was able to debug what was going on here. I would say this is a bug in our Gradle plugin.
Currently all the reports paths are shared among all the tasks (this will most likely be fixed by this https://github.com/detekt/detekt/pull/3687).

If you don't specify a path for them (your scenario), detekt will use the source-set name as default:
https://github.com/detekt/detekt/blob/48c1da8462d4b2d7e932b172751206d21ca96ed8/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/internal/DetektJvm.kt#L36-L39

So what happens in your case, is that the gradle plugin doesn't find a value for the reports filename specified, and uses `integrationTest` for it (this happens as detekt registers a `detektIntegrationTest` for it).

The reason why removing the `withType<Detekt> {}` lambda resolves the issue is that the plugin will lazily register a task. That configuration block we pass (the one I linked before) won't be executed till the task is actually needed. Once you call `withType<Detekt>` you force that lambda to be executed.

Long story short: as a workaround you can specify your own report destination to don't fallback on defaults

```kotlin
detekt {
  reports {
    html {
      enabled = true
      destination = file("build/reports/detekt/detekt.html")
    }
  }
}
```

As for detekt, we should definitely fix this issue in the next release.
